src/main/java/com/amihaiemil/docker/ReadLogString.java

```diff
@@ -0,0 +1,208 @@
+     */
+    ReadLogString(final ResponseHandler<HttpResponse> other) {
+        this.other = other;
+    }
+
+    @Override
+    public String handleResponse(final HttpResponse httpResponse)
+            throws IOException {
+        final HttpResponse resp = this.other.handleResponse(httpResponse);
+        return this.toString(resp.getEntity());
+    }
+
+    /**
+     * Docker logs contains header
+     * [8]byte{STREAM_TYPE, 0, 0, 0, SIZE1, SIZE2, SIZE3, SIZE4}
+     * STREAM_TYPE
+     * 0: stdin (is written on stdout)
+     * 1: stdout
+     * 2: stderr
+     *
+     * SIZE1, SIZE2, SIZE3, SIZE4 are the four bytes of the uint32 size
+     * encoded as big endian.
+     *
+     * This method do:
+     *
+     * 1) Read 8 bytes.
+     * 2) Choose stdout or stderr depending on the first byte.
+     * 3) Extract the frame size from the last four bytes.
+     * 4) Read the extracted size and output it on the correct output.
+     * 5) Goto 1.
+     *
+     * @param entity HttpEntity for read message.
+     * @return Logs from container in String.
+     * @throws IOException if the entity cannot be read
+     */
+    private String toString(final HttpEntity entity) throws IOException {
+        ContentType contentType = ContentType.get(entity);
+        InputStream instream = entity.getContent();
+        int capacity = this.getCapacity(entity);
+        CharArrayBuffer buffer = new CharArrayBuffer(capacity);
+        if (instream != null) {
+            try {
+                Charset charset = this.getCharset(contentType);
+                Reader reader = new InputStreamReader(instream, charset);
+                this.read(buffer, reader);
+            } finally {
+                instream.close();
+            }
+        }
+        return buffer.toString();
+    }
+
+    /**
+     * Read data from reader and save it in buffer.
+     *
+     * @param buffer Buffer for save message.
+     * @param reader Reader for read message.
+     * @throws IOException if the entity cannot be read
+     */
+    private void read(final CharArrayBuffer buffer,
+                      final Reader reader) throws IOException {
+        char[] controlChars = new char[8];
+        int len;
+        while (reader.read(controlChars) != -1) {
+            if (controlChars[0] != 0) {
+                long byteInLine = this.getUInt(controlChars);
+                char[] stdout;
+                if (byteInLine > 1024) {
+                    stdout = new char[1024];
+                } else {
+                    stdout = new char[(int) byteInLine];
+                }
+                while (byteInLine > 0) {
+                    len = reader.read(stdout);
+                    byteInLine -= len;
+                    if (len != -1) {
+                        buffer.append(stdout, 0, len);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Get content length of HttpEntity.
+     *
```

Comment:
@lumimul Same here, more details in the JavaDoc, please :)

Reply:
add more detail)